Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -0,0 +1,200 @@
+        setMobileCoreEnvironmentFileID(for: edgeEnvironment)
+        MobileCore.registerExtensions([Identity.self, Edge.self], {
+            print("Extensions registration is complete")
+            waitForRegistration.countDown()
+        })
+        XCTAssertEqual(DispatchTimeoutResult.success, waitForRegistration.await(timeout: 2))
+        
+        // Set Edge location hint value if one is set for the test target
+        if edgeLocationHint != nil {
+            print("Setting Edge location hint to: \(String(describing: edgeLocationHint?.rawValue))")
+            Edge.setLocationHint(edgeLocationHint?.rawValue)
+        }
+        else {
+            print("No preset Edge location hint is being used for this test.")
+        }
+    }
+    
+    public override func tearDown() {
+        super.tearDown()
+
+        // to revisit when AMSDK-10169 is available
+        // wait .2 seconds in case there are unexpected events that were in the dispatch process during cleanup
+        usleep(200000)
+        EventHub.reset()
+        UserDefaults.clearAll()
+        FileManager.default.clearCache()
+    }
+    
+    // MARK: - Upstream integration test cases
+    
+    func testSendEvent_withStandardExperienceEvent_receivesExpectedEventHandles() {
+        // Setup
+        // Test expectations that make sure the callbacks fire for:
+        // 1. The network response from Edge Network
+        // 2. The event response handles converted into events, captured by test case event listeners
+        let edgeRequestContentExpectation = XCTestExpectation(description: "Edge extension request content listener called")
+        let networkResponseExpectation = XCTestExpectation(description: "Network request callback called")
+        
+        // MARK: Network response assertions
+        testingDelegate.testCaseCompletion = { httpConnection in
+            print("SOURCE: testCaseCompletion: \(httpConnection)")
+            print("data as string: \(httpConnection.responseString)")
+            
+            httpConnection.response?.url
+            print("baseURL: \(httpConnection.response?.url?.host)")
+            if httpConnection.response?.url?.host == "obumobile5.data.adobedc.net" {
+                XCTAssertEqual(200, httpConnection.responseCode)
+                networkResponseExpectation.fulfill()
+            }
+            
+        }
+        
+        let validationJSON = #"""
+        {
+          "payload": [
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "Target",
+              "hint" : "35"
+            },
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "AAM",
+              "hint" : "9"
+            },
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "EdgeNetwork",
+              "hint" : "or2"
+            }
+          ]
+        }
+        """#
+        
+        // MARK: Response Event assertions
+        registerEdgeLocationHintListener() { event in
```

Comment:
Can this be extracted to function ex: ValidateLocationHintResponseHandleEvent().

So your test case can be
```
test {
 test api
 validateNetworkReq()
 validateResponseEvent()
}
```

Reply:
Or we could have a common function that can be used for registering listeners for a specific handle type, locationHint being one of them. 

We may need to also have support for testing multiple types and multiple handles for the same type in one test, or assert no other unexpected handles are received. 
e.g.:
```
expectHandlesWithType("locationHint:result", times: 1)
expectHandlesWithType("other:example", times: 2, expectedHandles: [{...}, {...}])
ignoreUnexpectedHandles(false)
```